src/test/java/edu/kis/powp/jobs2d/events/SelectTestFigureOptionListener.java

```diff
@@ -3,19 +3,29 @@
 import java.awt.event.ActionEvent;
 import java.awt.event.ActionListener;
 
+import edu.kis.powp.jobs2d.Job2dDriver;
 import edu.kis.powp.jobs2d.drivers.DriverManager;
 import edu.kis.powp.jobs2d.magicpresets.FiguresJoe;
 
 public class SelectTestFigureOptionListener implements ActionListener {
 
 	private DriverManager driverManager;
 
-	public SelectTestFigureOptionListener(DriverManager driverManager) {
+	private java.lang.reflect.Method method = null;
+
+	private FiguresJoe obj = new FiguresJoe();
+
+	public SelectTestFigureOptionListener(DriverManager driverManager, String figureScript) {
 		this.driverManager = driverManager;
+		try {
+		  method = obj.getClass().getMethod(figureScript, (Job2dDriver.class));
```

Comment:
Reflection is not object-oriented and type-safe mechanisms. Adding a new listener class would be consistent with an approach used in the project.

Reply:
I've chosen enum instead of listener. I don't now which option is better.

Reply:
Much better now :)